### DISCUSSION TITLE
Create Nx expo apps, with inherited base

### DISCUSSION THREAD

**Discussion by evgenyshenets** [nrwl/nx#23113] [General]
I need to have one base app, with initial stuff and I need to be able create lots of projects based on this shared(base) app.
Let's say, initial app with config and some code structure.
- initial;
  - app1 based on initial;
  - app2 based on initial;
  - ....

Does it make sense and how it will look like with nx?

**Comment by jaysoo** [comment]
@evgenyshenets Sounds like what you want is to create your own Nx plugin with a custom app generator. You could use the `@nx/expo:app` generator as a base to build on top of.

If the plugin/generator is only needed for one workspace, you can [create a local plugin](https://nx.dev/extending-nx/tutorials/create-plugin#create-a-local-plugin). Otherwise, if you want to publish and share this plugin/generator you can [create a published plugin](https://nx.dev/extending-nx/tutorials/publish-plugin).
